src/Servers/Kestrel/Core/src/Internal/Http2/Http2Connection.cs

```diff
@@ -106,6 +111,10 @@ public Http2Connection(HttpConnectionContext context)
             _serverSettings.HeaderTableSize = (uint)http2Limits.HeaderTableSize;
             _serverSettings.MaxHeaderListSize = (uint)httpLimits.MaxRequestHeadersTotalSize;
             _serverSettings.InitialWindowSize = (uint)http2Limits.InitialStreamWindowSize;
+
+            // Max number of streams we will accept is 2 times the max streams per connection.
```

Comment:
Not a huge deal since the min is going to be InitialStreamPoolSize (20) in most cases, but why would it ever be more than MaxStreamsPerConnection pooled? The only time there would be more than that many streams at once is when the client is aborting streams, right? That doesn't feel like a scenario where want to be heavily optimizing performance.

Reply:
Yeah this comment is a bit incorrect. The max we will ever go to is the MaxStreamPoolSize now, regardless of the initial size.